src/mono/mono/component/mini-wasm-debugger.c

```diff
@@ -370,6 +370,12 @@ mono_wasm_set_is_debugger_attached (gboolean is_attached)
 EMSCRIPTEN_KEEPALIVE gboolean 
 mono_wasm_send_dbg_command_with_parms (int id, MdbgProtCommandSet command_set, int command, guint8* data, unsigned int size, int valtype, char* newvalue)
 {
+	if (!debugger_enabled) {
+		EM_ASM ({
+			MONO.mono_wasm_add_dbg_command_received ($0, $1, $2, $3);
```

Comment:
What will this do? I'm wondering if this should simply `return TRUE`?
Just thinking aloud - is this kinda guard needed anywhere else? `mono_wasm_set_is_debugger_attached`?

Reply:
I would like to get the error on DebuggerProxy side, and these two commands will return an error if we add this `mono_wasm_add_dbg_command_received` statement.
`mono_wasm_set_is_debugger_attached `doesn't need this guard because it doesn't do anything on debugger-agent.

Again, this is not the correct fix, it's just a protection to avoid the assert. The correct fix should be on the linker side and @radekdoulik asked your opinion about it:
https://github.com/dotnet/runtime/issues/61917#issuecomment-975835249

Reply:
I think this is also using MONO from global namespace. Which I wanted to undo in https://github.com/dotnet/runtime/pull/61848/

Reply:
Maybe not, maybe this is run from some closure inside of emscripten ?